tools/fv_grid_tools.F90

```diff
@@ -586,7 +585,7 @@ subroutine init_grid(Atm, grid_name, grid_file, npx, npy, npz, ndims, nregions,
     if (Atm%flagstruct%grid_type>3) then
        if (Atm%flagstruct%grid_type == 4) then
           call setup_cartesian(npx, npy, Atm%flagstruct%dx_const, Atm%flagstruct%dy_const, &
-               Atm%flagstruct%deglat, Atm%bd)
+               Atm%flagstruct%deglat, Atm%flagstruct%domain_deg, Atm%bd, Atm)
```

Comment:
Because this is sending in elements of the Atm structure and the whole Atm, the interface should be simplified and remove the individual Atm elements as unexpected things can happen with scope INOUT variables.